src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -64,41 +74,36 @@ public override SourceText GetSubText(TextSpan span)
             int segOffset;
             GetIndexAndOffset(sourceIndex, out segIndex, out segOffset);
 
-            var newTexts = ArrayBuilder<SourceText>.GetInstance();
-            while (segIndex < _texts.Length && count > 0)
+            var newSegments = ArrayBuilder<SourceText>.GetInstance();
+            while (segIndex < _segments.Length && count > 0)
             {
-                var segment = _texts[segIndex];
+                var segment = _segments[segIndex];
                 var copyLength = Math.Min(count, segment.Length - segOffset);
 
-                AddSegments(newTexts, segment.GetSubText(new TextSpan(segOffset, copyLength)));
+                AddSegments(newSegments, segment.GetSubText(new TextSpan(segOffset, copyLength)));
 
                 count -= copyLength;
                 segIndex++;
                 segOffset = 0;
             }
 
-            if (newTexts.Count == 0)
-            {
-                newTexts.Free();
-                return SourceText.From(string.Empty, this.Encoding, this.ChecksumAlgorithm);
-            }
-            else if (newTexts.Count == 1)
-            {
-                SourceText result = newTexts[0];
-                newTexts.Free();
-                return result;
-            }
-            else
-            {
-                return new CompositeText(newTexts.ToImmutableAndFree());
-            }
+            return ToSourceTextAndFree(newSegments, this, adjustSegments: false);
         }
 
         private void GetIndexAndOffset(int position, out int index, out int offset)
         {
-            for (int i = 0; i < _texts.Length; i++)
+#if false
+            // TODO: not sure if this is necessary if number of segments is constrained.
```

Comment:
I guess my concern is a TODO inside of an #if false. I'd prefer you just pick one and delete the other code rather than have the indecision expressed here.